crates/map/src/helpers/astar.cairo

```diff
@@ -36,29 +37,34 @@ pub impl Astar of AstarTrait {
         let mut visited: Felt252Dict<bool> = Default::default();
         heap.add(start);
         // [Compute] Evaluate the path until the target is reached
-        while !heap.is_empty() {
+        while let Option::Some(current) = heap.pop_front() {
             // [Compute] Get the less expensive node
-            let current: Node = heap.pop_front().unwrap();
             visited.insert(current.position.into(), true);
             // [Check] Stop if we reached the target
             if current.position == target.position {
                 break;
             }
             // [Compute] Evaluate the neighbors for all 4 directions
-            if Self::check(grid, width, height, current.position, Direction::North, ref visited) {
-                let neighbor_position = current.position + width;
+            let seed = Seeder::shuffle(grid, current.position.into());
```

Comment:
Do we always want to shuffle here? Or this could be disabled?

Reply:
I don't see any use case where we want it to be disabled, I assume anyone is using a pathfinder expect it to not have a pattern (like always going north if possible, then east, etc.). The way the seed is used ensure reproducibility, so I feel comfortable as is.